# HG changeset patch
# User click
# Date 1278269864 -7200
# Branch trunk
# Node ID 4a465394b47fe21db7eb9c952d7fe576aed233aa
# Parent  3d94318355eb680f44e234060ea8532f05482fa2
resolve circular dependency on uninitialized memory (and add copyrightheaders) in AHBot
(dependency-fix by ge0rg)

diff -r 3d94318355eb -r 4a465394b47f src/server/game/AuctionHouse/AuctionHouseBot/AuctionHouseBot.cpp
--- a/src/server/game/AuctionHouse/AuctionHouseBot/AuctionHouseBot.cpp	Sun Jul 04 20:17:22 2010 +0200
+++ b/src/server/game/AuctionHouse/AuctionHouseBot/AuctionHouseBot.cpp	Sun Jul 04 20:57:44 2010 +0200
@@ -1,10 +1,27 @@
+/*
+ * Copyright (C) 2008-2010 Trinity <http://www.trinitycore.org/>
+ * Copyright (C) 2005-2009 MaNGOS <http://getmangos.com/>
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+ */
+
 #include "ObjectMgr.h"
 #include "AuctionHouseMgr.h"
 #include "AuctionHouseBot.h"
 #include <vector>
 
-
-
 using namespace std;
 vector<uint32> npcItems;
 vector<uint32> lootItems;
@@ -22,6 +39,7 @@
 vector<uint32> purpleItemsBin;
 vector<uint32> orangeItemsBin;
 vector<uint32> yellowItemsBin;
+
 AuctionHouseBot::AuctionHouseBot()
 {
     debug_Out = false;
@@ -1509,6 +1527,7 @@
             uint32 maxItems = (uint32) strtoul(param1, NULL, 0);
             CharacterDatabase.PExecute("UPDATE auctionhousebot SET maxitems = '%u' WHERE auctionhouse = '%u'", maxItems, ahMapID);
             config->SetMaxItems(maxItems);
+            config->CalculatePercents();
         }
         break;
     case 3:     //min time Deprecated (Place holder for future commands)
diff -r 3d94318355eb -r 4a465394b47f src/server/game/AuctionHouse/AuctionHouseBot/AuctionHouseBot.h
--- a/src/server/game/AuctionHouse/AuctionHouseBot/AuctionHouseBot.h	Sun Jul 04 20:17:22 2010 +0200
+++ b/src/server/game/AuctionHouse/AuctionHouseBot/AuctionHouseBot.h	Sun Jul 04 20:57:44 2010 +0200
@@ -1,3 +1,22 @@
+/*
+ * Copyright (C) 2008-2010 Trinity <http://www.trinitycore.org/>
+ * Copyright (C) 2005-2009 MaNGOS <http://getmangos.com/>
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+ */
+
 #ifndef AUCTION_HOUSE_BOT_H
 #define AUCTION_HOUSE_BOT_H
 
@@ -173,7 +192,8 @@
     void SetMaxItems(uint32 value)
     {
         maxItems = value;
-        CalculatePercents();
+	// CalculatePercents() needs to be called, but only if
+	// SetPercentages() has been called at least once already.
     }
     uint32 GetMaxItems()
     {
